macros/src/lib.rs

```diff
@@ -19,27 +19,35 @@ pub fn type_code(args: TokenStream, input: TokenStream) -> TokenStream {
     let mut variants = Vec::new();
     let mut decodes = Vec::new();
     let mut encodes = Vec::new();
+    let mut enumerator = 0;
 
-    for (index, variant) in input.variants.iter().enumerate() {
+    for variant in input.variants.iter() {
         let camel = &variant.ident;
         let camel_name = camel.to_string();
 
-        if camel_name != "not_used" {
-            let snake = syn::Ident::new(&to_snake(&camel_name), camel.span());
-            let index = index as u32;
+        if let Some((_, value)) = &variant.discriminant {
+            if let syn::Expr::Lit(value) = value {
```

Comment:
You can inline this `if let` into the one above:

`if let Some((_, syn::Expr::Lit(value))) = &variant.discriminant {`